cli/command/service_test.go

```diff
@@ -0,0 +1,322 @@
+
+	base.Resolver.EXPECT().
+		Resolve("service", "svc_name").
+		Return([]string{"svc_id"}, nil)
+
+	base.Client.EXPECT().
+		ReadServiceLogs("svc_id", url.Values{
+			"tail":  []string{"100"},
+			"start": []string{"start"},
+			"end":   []string{"end"},
+		})
+
+	flags := map[string]interface{}{
+		"tail":  100,
+		"start": "start",
+		"end":   "end",
+	}
+
+	c := getCLIContext(t, []string{"svc_name"}, flags)
+	if err := serviceCommand.logs(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestServiceLogs_userInputError(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	serviceCommand := NewServiceCommand(base.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": getCLIContext(t, nil, nil),
+	}
+
+	for name, c := range contexts {
+		if err := serviceCommand.logs(c); err == nil {
+			t.Fatal("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestReadService(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	serviceCommand := NewServiceCommand(base.Command())
+
+	base.Resolver.EXPECT().
+		Resolve("service", "svc_name").
+		Return([]string{"svc_id"}, nil)
+
+	base.Client.EXPECT().
+		ReadService("svc_id").
+		Return(&models.Service{}, nil)
+
+	c := getCLIContext(t, []string{"svc_name"}, nil)
+	if err := serviceCommand.read(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestReadService_userInputError(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	serviceCommand := NewServiceCommand(base.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": getCLIContext(t, nil, nil),
+	}
+
+	for name, c := range contexts {
+		if err := serviceCommand.read(c); err == nil {
+			t.Fatal("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestScaleService(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	serviceCommand := NewServiceCommand(base.Command())
+
+	scale := 2
+	req := models.UpdateServiceRequest{
+		ServiceID: "svc_id",
+		Scale:     &scale,
+	}
+
+	base.Resolver.EXPECT().
+		Resolve("service", "svc_name").
+		Return([]string{"svc_id"}, nil)
+
+	base.Client.EXPECT().
+		UpdateService(req).
+		Return("job_id", nil)
+
+	base.Client.EXPECT().
+		ReadJob("job_id").
+		Return(&models.Job{
+			Status: "Completed",
+			Result: "svc_id",
+		}, nil)
+
+	base.Client.EXPECT().
+		ReadService("svc_id").
+		Return(&models.Service{}, nil)
+
+	c := getCLIContext(t, []string{"svc_name", "2"}, nil)
+	if err := serviceCommand.scale(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestScaleService_userInputError(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	serviceCommand := NewServiceCommand(base.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg":  getCLIContext(t, nil, nil),
+		"Missing COUNT arg": getCLIContext(t, []string{"svc_name"}, nil),
+	}
+
+	for name, c := range contexts {
+		if err := serviceCommand.scale(c); err == nil {
+			t.Fatal("%s: error was nil!", name)
+		}
+	}
```

Comment:
I was made aware of `t.Run()` in my PR for these types of tests that unlocks some benefits when running the tests. The structure would look like:
```
	for name, c := range contexts {
		t.Run(name, func(t *testing.T) {
			if err := serviceCommand.scale(c); err == nil {
				t.Fatal("%s: error was nil!", name)
			}
		})
	}
```
You will get breakdowns of each iteration of `contexts` that will look like:
```
=== RUN   TestParsePortErrors
=== RUN   TestParsePortErrors/Non-integer_CONTAINER_PORT
=== RUN   TestParsePortErrors/Missing_HOST_PORT
=== RUN   TestParsePortErrors/Missing_CONTAINER_PORT
=== RUN   TestParsePortErrors/Missing_PROTOCOL
=== RUN   TestParsePortErrors/Non-integer_HOST_PORT
--- PASS: TestParsePortErrors (0.00s)
    --- PASS: TestParsePortErrors/Non-integer_CONTAINER_PORT (0.00s)
    --- PASS: TestParsePortErrors/Missing_HOST_PORT (0.00s)
    --- PASS: TestParsePortErrors/Missing_CONTAINER_PORT (0.00s)
    --- PASS: TestParsePortErrors/Missing_PROTOCOL (0.00s)
    --- PASS: TestParsePortErrors/Non-integer_HOST_PORT (0.00s)
```

Reply:
Oh yeah, great note! I completely forgot about `t.Run()`, but now I recall that you and I had used it way back when we were poking around with benchmarking. 👍

Reply:
✅